winch/codegen/src/abi/mod.rs

```diff
@@ -118,178 +125,310 @@ pub(crate) trait ABI {
     /// calling convention.
     fn callee_saved_regs(call_conv: &CallingConvention) -> SmallVec<[(Reg, OperandSize); 18]>;
 
-    /// Returns the size of each argument stack slot per argument type.
-    fn stack_arg_slot_size_for_type(ty: WasmType) -> u32;
+    /// The size, in bytes, of each stack slot used for stack parameter passing.
+    fn stack_slot_size() -> u32;
+
+    /// Returns the size in bytes of the given [`WasmType`].
+    fn sizeof(ty: &WasmType) -> u32 {
+        match ty {
+            WasmType::Ref(rt) => match rt.heap_type {
+                WasmHeapType::Func => Self::word_bytes(),
+                ht => unimplemented!("Support for WasmHeapType: {ht}"),
+            },
+            WasmType::F64 | WasmType::I64 => Self::word_bytes(),
+            WasmType::F32 | WasmType::I32 => Self::word_bytes() / 2,
```

Comment:
This seems like it is unnecessarily setting us up for porting issues in the future when we have a 32-bit target. Can we just put the size directly or use `mem::size_of::<u32>()` or something instead?

Reply:
I ended up moving the definition to the implementation of each ABI, that way once it'll be easier to integrate once we support 32-bit targets.